stratum/hal/lib/common/target_options.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef STRATUM_HAL_LIB_COMMON_TARGET_OPTIONS_H_
+#define STRATUM_HAL_LIB_COMMON_TARGET_OPTIONS_H_
+
+namespace stratum {
+namespace hal {
+
+class TargetOptions {
+ public:
+  TargetOptions();
+
+  // Whether ConfigMonitoringService should push the updated chassis
+  // configuration.
+  bool pushUpdatedChassisConfig;
+
+  // Whether ConfigMonitoringService should require that the config file
+  // be a regular file (not a symlink).
+  bool secureChassisConfig;
```

Comment:
Why/when would we allow a symlink ChassisConfig file? What is the use case for this?

Reply:
> Why/when would we allow a symlink ChassisConfig file? What is the use case for this?

Two use cases:

1. One of the unit tests uses a symlink.
2. We need to disable the symlink check for non-TDI targets, because it changes the behavior of a existing code. The check must be conditionalized somehow, and I'm trying to do it without #ifdefs, which are a pain in Bazel.